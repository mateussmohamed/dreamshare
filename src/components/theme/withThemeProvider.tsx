import React, { Component, FC } from 'react'
import { ThemeProvider } from 'styled-components'

import theme from './index'

const getDisplayName = (component: FC) =>
  component.displayName || component.name || 'Component'

const withThemeProvider = (WrappedComponent: FC) =>
  class extends Component {
    static displayName = `withThemeProvider(${getDisplayName(
      WrappedComponent,
    )})`

    render() {
      return (
        <ThemeProvider theme={theme}>
          <WrappedComponent {...this.props} />
        </ThemeProvider>
      )
    }
  }

export default withThemeProvider
