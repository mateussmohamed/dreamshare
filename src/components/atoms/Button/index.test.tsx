import React from 'react'

import { Button } from './index'
import withThemeProvider from '../../theme/withThemeProvider'

describe('[<Button />]', () => {
  const wrap = props => withThemeProvider(() => <Button {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({ children: 'I am a Title' })

    expect(wrapper).toMatchSnapshot()
  })

  describe('when pass props', () => {
    test('primary', () => {
      const wrapper = wrap({ children: 'Primay Button', primary: true })

      expect(wrapper).toMatchSnapshot()
    })

    test('secondary', () => {
      const wrapper = wrap({
        children: 'Secondary Button',
        secondary: true,
      })

      expect(wrapper).toMatchSnapshot()
    })

    test('ghost', () => {
      const wrapper = wrap({ children: 'Ghot Button', ghost: true })

      expect(wrapper).toMatchSnapshot()
    })
  })
})
