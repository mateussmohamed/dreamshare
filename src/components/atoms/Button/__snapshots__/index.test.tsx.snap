// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Button />] matches to snapshot 1`] = `[Function]`;

exports[`[<Button />] when pass props ghost 1`] = `[Function]`;

exports[`[<Button />] when pass props primary 1`] = `[Function]`;

exports[`[<Button />] when pass props secondary 1`] = `[Function]`;
